<a href="https://colab.research.google.com/github/leenago/Self_studying/blob/master/Statistics/statistics_9%EC%9E%A5.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# 9장. 신뢰성 분석

##1.준비하기

### 1)Google Drive와 연동

In [ ]:
from google.colab import drive
drive.mount('/content/gdrive')

Drive already mounted at /content/gdrive; to attempt to forcibly remount, call drive.mount("/content/gdrive", force_remount=True).


In [ ]:
%cd /content/gdrive/My Drive/파비통 데이터

/content/gdrive/My Drive/파비통 데이터


In [ ]:
%ls

'실습파일_10장 분산분석.ipynb'
'실습파일_11장 회귀분석.ipynb'
'실습파일_12장 요인분석.ipynb'
'실습파일_13장 분류예측분석.ipynb'
'실습파일_14장 군집분석.ipynb'
'실습파일_15장 포지셔닝 분석.ipynb'
'실습파일_16장 컨조인트 분석.ipynb'
'실습파일_17장 비모수 통계분석.ipynb'
'실습파일_3장 파이썬 기초.ipynb'
'실습파일_4장 데이터전처리 및 기초분석.ipynb'
'실습파일_5장 기술통계분석.ipynb'
'실습파일_6장 t 검정.ipynb'
'실습파일_7장 상관관계분석.ipynb'
'실습파일_8장 범주형 데이터 분석.ipynb'
'실습파일_9장 신뢰성 분석.ipynb'
 Ashopping.csv
 CCA.csv
 Conjoint.csv
 Correspondence.csv
 부록_통계표.docx
 MDS1.csv
 MDS3.csv
 주택+선호도+조사.sav
'Step by Step 파이썬 비즈니스 통계분석_정오표.xlsx'


### 2) 필요 모듈 불러오기

In [ ]:
!pip install pingouin

In [ ]:
import pandas as pd
import pingouin as pg
#pingouin: pandas, numpy를 기반으로 한 다양한 통계기능 제공 패키지

##2.내적일관성 분석 과정

### 1) 파이썬 비즈니스 통계분석에서 주어진 데이터

In [ ]:
df = pd.read_csv('Ashopping.csv',sep=',',encoding='CP949')

df[['친절성', '신속성', '책임성', '정확성', '전문성']].head()
#수치형 데이터인 것을 볼 수 있다.

친절성  신속성  책임성  정확성  전문성
0    6    6    6    6    6
1    5    3    6    6    6
2    7    7    6    6    7
3    6    6    6    5    6
4    5    6    6    5    6

In [ ]:
X = df[['친절성', '신속성', '책임성', '정확성', '전문성']]

In [ ]:
#크론바흐 알파 계수 출력
pg.cronbach_alpha(data=X)
#data: 분석의 대상이 되는 데이터 입력하는 parameter

(0.8117458883194828, array([0.793, 0.83 ]))

- 크론바흐 알파 계수
  - *0.7* 이상: 신뢰성이 **높다**.

- *'[0.793, 0.83]'* : 신뢰수준 95%의 신뢰구간

#### 각 변수를 모두 사용하는 것의 적절성 여부 판단

In [ ]:
X1 = df[['신속성', '책임성', '정확성', '전문성']] #친절성 제외
X2 = df[['친절성', '책임성', '정확성', '전문성']] #신속성 제외
X3 = df[['친절성', '신속성', '정확성', '전문성']] #책임성 제외
X4 = df[['친절성', '신속성', '책임성', '전문성']] #정확성 제외
X5 = df[['친절성', '신속성', '책임성', '정확성']] #전문성 제외

In [ ]:
print('X1')
print(pg.cronbach_alpha(data=X1))
print('*'*45)
print('X2')
print(pg.cronbach_alpha(data=X2))
print('*'*45)
print('X3')
print(pg.cronbach_alpha(data=X3))
print('*'*45)
print('X4')
print(pg.cronbach_alpha(data=X4))
print('*'*45)
print('X5')
print(pg.cronbach_alpha(data=X5))
print('*'*45)

X1
(0.8057989412964606, array([0.785, 0.825]))
*********************************************
X2
(0.7836410115861367, array([0.761, 0.805]))
*********************************************
X3
(0.7811976111855439, array([0.758, 0.803]))
*********************************************
X4
(0.7713180205968039, array([0.747, 0.794]))
*********************************************
X5
(0.7341054622297101, array([0.706, 0.76 ]))
*********************************************


- 각 변수를 제외했을 때의 크론바흐 알파계수는, 모든 변수를 사용했을 때의 크론바흐 알파 계수(**0.811**)보다 낮은 값을 보인다.

  → 신뢰성 분석 결과, '5가지 변수를 모두 사용하는 것이 가장 적절한 직원 만족도 평가를 이끌어낸다' 라고 결론지을 수 있음. 

### 2) 외부 데이터 사용
- [데이터 출처: 보건과학통계 SPSS 이야기](http://blog.naver.com/PostView.nhn?blogId=y4769&logNo=220313366439)
  - 데이터 내용: '**병원 종사자들은 어떤 기준으로 주택을 선택하는가**'에 대한 설문조사

####(1)데이터 불러오기

In [ ]:
#주어진 데이터가 SPSS 파일이기 때문에, 이를 변환해줄 수 있는 패키지를 설치한다.
!pip install pyreadstat

In [ ]:
df2 = pd.read_spss('주택+선호도+조사.sav'); df2

fa1   fa2      fa3      fa4      fa5      fa6     fa7
0    2.66667   3.5  3.66667  3.33333      그렇다      그렇다  매우 그렇다
1    3.33333   그렇다  3.66667  2.33333      그렇다   그렇지 않다     그렇다
2       보통이다  보통이다  3.33333  2.66667  2.66667      그렇다    보통이다
3    3.33333  보통이다  3.33333   그렇지 않다  4.33333  3.66667     3.5
4     그렇지 않다  보통이다  2.66667  2.33333      그렇다  3.33333     3.5
..       ...   ...      ...      ...      ...      ...     ...
195      그렇다   그렇다      그렇다     보통이다      그렇다      그렇다  매우 그렇다
196     보통이다   그렇다      그렇다     보통이다      그렇다      그렇다     그렇다
197  3.33333   그렇다      그렇다  2.33333   그렇지 않다      그렇다     그렇다
198  3.66667   그렇다      그렇다   그렇지 않다  3.66667  3.66667     그렇다
199     보통이다  보통이다     보통이다     보통이다      그렇다     보통이다     3.5

[200 rows x 7 columns]

####(2)데이터 전처리

In [ ]:
df2 = df2.rename(columns={'fa1':'주택크기','fa2':'주택방향','fa3':'자녀학군','fa4':'통근거리','fa5':'교통편리','fa6':'주변환경','fa7':'브랜드'})
df2.head()

주택크기  주택방향     자녀학군     통근거리     교통편리     주변환경     브랜드
0  2.66667   3.5  3.66667  3.33333      그렇다      그렇다  매우 그렇다
1  3.33333   그렇다  3.66667  2.33333      그렇다   그렇지 않다     그렇다
2     보통이다  보통이다  3.33333  2.66667  2.66667      그렇다    보통이다
3  3.33333  보통이다  3.33333   그렇지 않다  4.33333  3.66667     3.5
4   그렇지 않다  보통이다  2.66667  2.33333      그렇다  3.33333     3.5

In [ ]:
new_df2 = df2.replace({'전혀 그렇지 않다':1,'그렇지 않다':2,'보통이다':3,'그렇다':4,'매우 그렇다':5}).astype(int)
new_df2

주택크기  주택방향  자녀학군  통근거리  교통편리  주변환경  브랜드
0       2     3     3     3     4     4    5
1       3     4     3     2     4     2    4
2       3     3     3     2     2     4    3
3       3     3     3     2     4     3    3
4       2     3     2     2     4     3    3
..    ...   ...   ...   ...   ...   ...  ...
195     4     4     4     3     4     4    5
196     3     4     4     3     4     4    4
197     3     4     4     2     2     4    4
198     3     4     4     2     3     3    4
199     3     3     3     3     4     3    3

[200 rows x 7 columns]

In [ ]:
new_df2.columns

Index(['주택크기', '주택방향', '자녀학군', '통근거리', '교통편리', '주변환경', '브랜드'], dtype='object')

####(3)내적일관성 분석하기

In [ ]:
#모든 변수를 포함한 크론바흐 알파 계수 출력
X = new_df2[['주택크기', '주택방향', '자녀학군', '통근거리', '교통편리', '주변환경', '브랜드']]

pg.cronbach_alpha(data=X)

(0.7622011986184487, array([0.708, 0.809]))

In [ ]:
#각 변수를 제외한 크론바흐 알파 계수 출력

X1 = new_df2[['주택방향', '자녀학군', '통근거리', '교통편리', '주변환경', '브랜드']] #주택크기 제외
X2 = new_df2[['주택크기', '자녀학군', '통근거리', '교통편리', '주변환경', '브랜드']] #주택방향 제외
X3 = new_df2[['주택크기', '주택방향', '통근거리', '교통편리', '주변환경', '브랜드']] #자녀학군 제외
X4 = new_df2[['주택크기', '주택방향', '자녀학군', '교통편리', '주변환경', '브랜드']] #통근거리 제외
X5 = new_df2[['주택크기', '주택방향', '자녀학군', '통근거리', '주변환경', '브랜드']] #교통편리 제외
X6 = new_df2[['주택크기', '주택방향', '자녀학군', '통근거리', '교통편리', '브랜드']] #주변환경 제외
X7 = new_df2[['주택크기', '주택방향', '자녀학군', '통근거리', '교통편리', '주변환경']] #브랜드 제외

In [ ]:
print('X1')
print(pg.cronbach_alpha(X1))
print('*'*45)
print('X2')
print(pg.cronbach_alpha(X2))
print('*'*45)
print('X3')
print(pg.cronbach_alpha(X3))
print('*'*45)
print('X4')
print(pg.cronbach_alpha(X4))
print('*'*45)
print('X5')
print(pg.cronbach_alpha(X5))
print('*'*45)
print('X6')
print(pg.cronbach_alpha(X6))
print('*'*45)
print('X7')
print(pg.cronbach_alpha(X7))
print('*'*45)

X1
(0.7259898609698825, array([0.663, 0.781]))
*********************************************
X2
(0.7200059501219583, array([0.655, 0.776]))
*********************************************
X3
(0.7400694272253906, array([0.68 , 0.792]))
*********************************************
X4
(0.7331791870187037, array([0.672, 0.787]))
*********************************************
X5
(0.7083506944790559, array([0.641, 0.767]))
*********************************************
X6
(0.7863046881923723, array([0.737, 0.829]))
*********************************************
X7
(0.7136453717328252, array([0.647, 0.771]))
*********************************************


In [ ]:
#변수를 5개로 주었을 때의 크론바흐 알파 계수
X8 = new_df2[['주택크기', '주택방향', '교통편리','통근거리','브랜드']] 
pg.cronbach_alpha(X8)

(0.777601909620842, array([0.725, 0.823]))



---


- *X6*의 크론바흐 알파 계수가 **0.786**으로 가장 높았다.
      - X6: '주택크기', '주택방향', '자녀학군', '통근거리', '교통편리', '브랜드' / '주변환경' 제외


---



- 변수를 5개로 주었을 때(*X8*), 가장 높게 나왔던 크론바흐 알파 계수는 **0.778** 이다.
        - X8: '주택크기', '주택방향', '교통편리','통근거리','브랜드'
  - 이는 X6의 크론바흐 알파 계수(**0.708**) 다음으로 높은 수치이다. 
  - 통상 신뢰도는 항목이 많을수록, 척도점이 많을수록 높아진다.
  - 하지만 변수가 5개임에도 불구하고, X8 변수 간의 크론바흐 알파 계수가 X6을 제외한 나머지의 크론바흐 알파 계수보다 높은 것으로 미루어보아, 이 변수 간의 상관관계가 매우 높다는 것을 알 수 있다.
---


→  '주택크기', '주택방향', '자녀학군', '통근거리', '교통편리', '브랜드'를 사용하여 설문지의 신뢰도를 높이는 것이 적절하다.